### ISSUE TITLE
Warnings on latest ecto

### ISSUE THREAD

**Issue by axelson** [onboardingsystems/enum_type#9]
Ecto 3.2.0 added the `embed_as/1` callback to the `Ecto.Type` behaviour:

https://github.com/elixir-ecto/ecto/blob/master/CHANGELOG.md#enhancements-11

So this causes a warning when used with ecto >= 3.2.0:

```
warning: function embed_as/1 required by behaviour Ecto.Type is not implemented (in module MyApp.MyEnum)
  lib/my_app/my_enum.ex:8: MyApp.MyEnum (module)

warning: function equal?/2 required by behaviour Ecto.Type is not implemented (in module MyApp.MyEnum)
  lib/my_app/my_enum.ex:8: MyApp.MyEnum (module)
```

It would be great to implement those callbacks (I don't see in what release `equal?/2` was added). But actually in my case I'd like to not generate an Ecto type for this enum. Maybe I'll take a stab at a PR that makes generating the ecto type optional.

**Comment by brettv** [comment]
Yes, I noticed those warnings too when I did a trial install of a newer Ecto version.

Thank you for the PR.  I will have someone on my team review it.  Thanks!

**Comment by axelson** [comment]
Great, thanks!

**Comment by walu-lila** [comment]
@brettv [This conversation](https://elixirforum.com/t/please-explain-ecto-type-embed-as-1/26552/3) indicates that if you replace `@behaviour Ecto.Type` with `use Ecto.Type` in your type module you will get a default implementations of `embed_as/1` and `equal?/2`.

This would remove the warnings

**Comment by jtrees** [comment]
Any progress on this?

**Comment by cadebward** [comment]
Thanks @walu-lila for the link. I've gone ahead and made those changes and can confirm that the compiler warnings are now gone. We also finally got access to the hex account so I've published a new version with these changes: v1.1.2. Let me know if any of you have issues.
